### DISCUSSION TITLE
Segmnention Fault

### DISCUSSION THREAD

**Discussion by SantiagoOsorio3** [IIC2133-PUC/2024-2#80] [Tarea 1] [answered]
Hola, tengo una duda ya que mi programa esta arrojando un segmentio fault, sin embargo cuando lo corro en valgrind me sale 0 errores. Alguien sabe alguna forma de saber donde puede estar el error..?

Gracias.

**Comment by sebacarrascop** [comment, marked as answer]
Hola, es posible que cuando dices:

> sin embargo cuando lo corro en Valgrind me sale 0 errores

no sea del todo correcto. Te recomiendo leer en detalle lo que dice Valgrind, ya que aunque no reporte errores graves, puede estar mostrando advertencias u otros detalles importantes. Para obtener más información, puedes usar flags adicionales como estas:
`
--leak-check=full --show-leak-kinds=all --track-origins=yes
`

La razón por la que parece que tu programa "corre" con valgrind es porque valgrind, al encontrar un error, intentará continuar la ejecución para detectar todos los posibles problemas, por lo que incluso si tu programa parece correr bien con Valgrind, el error puede seguir ocurriendo en una ejecución normal. Es posible que Valgrind "oculte" o maneje el error de alguna manera.

**Reply by SantiagoOsorio3** [reply]
perfecto! igualmente lo corri con  --leak-check=full y --show-leak-kinds=all,  en donde en ambos casos me dio :

valgrind --show-leak-kinds=all ./program input.txt output.txt
==94== Memcheck, a memory error detector
==94== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==94== Using Valgrind-3.15.0 and LibVEX; rerun with -h for copyright info
==94== Command: ./program input.txt output.txt
==94== 
ENTER
ENTER
ENTER
ENTER
ENTER
LEAVE
FESTIVAL
HAPPINESS-BUFF
1
==94== 
==94== HEAP SUMMARY:
==94==     in use at exit: 0 bytes in 0 blocks
==94==   total heap usage: 14 allocs, 14 frees, 6,392 bytes allocated
==94== 
==94== All heap blocks were freed -- no leaks are possible
==94== 
==94== For lists of detected and suppressed errors, rerun with: -s
==94== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
root@9806622cfb78:~/volume/probar_setup# 

Nose si es que soy yo, pero no encuentro ninguna advertencia
